### PR TITLE
Add documentation for JSON output

### LINKED ISSUE #637: Document JSON output
Semgrep can output --json, but we haven't documented the JSON format and fields. This would be smart. We've already done most of it for the r2c analyzer json output spec.

### PR DESCRIPTION
**Pull request by mschwager**
Fix #637.

For the rendered text see https://github.com/returntocorp/semgrep/blob/222edab31377ccb7a306ed75907ed664e0011da3/docs/output.md

### CODE REVIEW

**Review by ievans — APPROVED** [review]

File: docs/output.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+* [JSON](#json)
+
+## Default
+
+By default, `semgrep` outputs results to `stdout`. The output looks like:
+
+```
+<finding-file-path>
+rule:<rule-id>: <rule-message>
+<finding-line-number>: <finding-line-code>
```
Comment (ievans) [inline docs/output.md:19]: Is this actually helpful to document? I'm on the fence.

File: docs/SUMMARY.md [hunk 2]
```diff
@@ -4,6 +4,7 @@
 * [Live Editor](https://semgrep.live)
 * [Pattern Features](pattern-features.md)
 * [Configuration Files](configuration-files.md)
+* [Output](output.md)
 * [Languages](matrix.md)
 * [Integrations](integrations.md)
 * [Development](development.md)
```
Comment (ievans) [inline docs/SUMMARY.md:10]: @DrewDennison / @mschwager  -- how are people finding this summary.md doc anyways? Should we link to the output.md from the main readme?

File: docs/output.md (hunk 1 above)
Comment (mschwager) [reply docs/output.md:19]: Yup, I agree! I don't have strong feelings either way.

File: docs/SUMMARY.md (hunk 2 above)
Comment (mschwager) [reply docs/SUMMARY.md:10]: Good question - I think we need to have a broader discussion on where we should be directing people for documentation. Are we set on GitBook? Is Github + Markdown sufficient? Something else? Currently we direct people to the Github + Markdown docs, but the GitBook is also available: https://docs.semgrep.dev/. I've mostly been creating content and delaying that decision - maybe @pabloest has thoughts too?

Comment (pabloest) [reply docs/SUMMARY.md:10]: I have not looked into writing/publishing with GitBook, but as a reader I find it very pleasing to consume and digest. That may be just the theme @mschwager chose and something similar would be possible using Readthedocs. 

For writing and deploying docs I found Readthedocs a bit clunky but I only used it a few times.

In the short term we should definitely link to the docs and in the longer term we could consider integrating GitBook (or whatever we choose) into our website.

File: docs/output.md (hunk 1 above)
Comment (mschwager) [reply docs/output.md:19]: I'll leave it in for now - we can always iterate on the doc itself and the broader documentation discussion.

File: docs/SUMMARY.md (hunk 2 above)
Comment (mschwager) [reply docs/SUMMARY.md:10]: @DrewDennison pulled in GitBook initially. I'm not opposed to it, but I don't generally think we have a cohesive documentation story at the moment. At some point we'll probably want to be consistent with things like Bento and other projects :+1:

**Pull request merged**